app/src/main/java/com/gh4a/fragment/CommitListFragment.java

```diff
@@ -92,13 +108,55 @@ public void onCreate(Bundle savedInstanceState) {
         mRepoName = getArguments().getString("repo");
         mRef = getArguments().getString("ref");
         mFilePath = getArguments().getString("path");
+        mFileType = (ContentType) getArguments().getSerializable("type");
+        if (savedInstanceState != null) {
+            mFollowRenames = savedInstanceState.getBoolean("follow_renames");
+            mMostRecentRenameFollowData = savedInstanceState.getParcelable("rename_follow_data");
+            mOldestLoadedCommit = savedInstanceState.getParcelable("oldest_commit");
+        }
+    }
+
+    @Override
+    public void onSaveInstanceState(@NonNull Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putBoolean("follow_renames", mFollowRenames);
+        outState.putParcelable("rename_follow_data", mMostRecentRenameFollowData);
+        outState.putParcelable("oldest_commit", mOldestLoadedCommit);
+    }
+
+    @Override
+    public void onCreateMenu(@NonNull Menu menu, @NonNull MenuInflater inflater) {
+        if (mFileType == ContentType.File) {
```

Comment:
```suggestion
        if (mFileType == ContentType.File || mFileType == ContentType.Submodule) {
```
I've just checked with submodules [in this folder](https://github.com/topjohnwu/Magisk/tree/master/native/src/external) and following renames works fine for them too.
At the moment this change wouldn't be strictly necessary since submodules are returned with their type set as `file`, but in a future REST API version GH might change that according to the [API docs](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#get-repository-content):
> In the next major version of the API, the type will be returned as "submodule".